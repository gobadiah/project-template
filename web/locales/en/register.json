{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
