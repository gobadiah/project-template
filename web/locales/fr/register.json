{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
