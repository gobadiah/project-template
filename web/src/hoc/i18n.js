import i18n from '~/config/i18n';

export default ({ req, namespaces }) => new Promise((resolve) => {
  if (req) {
    resolve(i18n.getInitialProps(req, namespaces));
  } else {
    resolve({});
  }
});
