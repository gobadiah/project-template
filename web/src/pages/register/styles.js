import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
