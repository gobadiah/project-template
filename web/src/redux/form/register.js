import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
