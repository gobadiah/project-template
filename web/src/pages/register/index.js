import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
