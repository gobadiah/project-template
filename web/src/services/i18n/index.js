import i18n from 'i18next';
import backend from 'i18next-xhr-backend';
import LanguageDetector from 'i18next-browser-languagedetector';

export const availableLanguages = ['fr', 'en'];
export const availableNamespaces = ['common', 'index'];

const detection = {
  order: ['querystring', 'cookie', 'localStorage', 'navigator', 'header'],

  lookupQuerystring: 'lng',
  lookupCookie: 'i18next',
  lookupLocalStorage: 'i18nextLng',

  caches: false,
};

const options = {
  fallbackLng: 'fr',
  load: 'languageOnly',

  ns: availableNamespaces,
  defaultNS: 'common',

  debug: false,

  react: {
    wait: true,
  },

  saveMissing: true,

  detection,

  interpolation: {
    escapeValue: false,
  },
};

if (process.browser) {
  i18n.use(backend)
    .use(LanguageDetector);
  options.caches = ['localStorage', 'cookie'];
  options.backend = {
    loadPath: '/locales/{{lng}}/{{ns}}.json',
    addPath: '/locales/add/{{lng}}/{{ns}}',
  };
}

if (!i18n.isInitialized) {
  i18n.init(options);
}

i18n.getInitialProps = (req, nss) => {
  let namespaces = nss;
  if (typeof namespaces === 'undefined') {
    namespaces = i18n.options.defaultNS;
  }
  if (typeof namespaces === 'string') {
    namespaces = [namespaces];
  }

  req.i18n.toJSON = () => null;

  const initialI18nStore = {};
  req.i18n.languages.forEach((l) => {
    initialI18nStore[l] = {};
    namespaces.forEach((ns) => {
      initialI18nStore[l][ns] = req.i18n.services.resourceStore.data[l] ?
        req.i18n.services.resourceStore.data[l][ns] || {} : {};
    });
  });

  return {
    i18n: req.i18n,
    initialI18nStore,
    initialLanguage: req.i18n.language,
  };
};

export default i18n;
