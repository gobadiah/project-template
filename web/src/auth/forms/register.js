import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
