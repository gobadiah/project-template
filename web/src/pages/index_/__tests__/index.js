import 'jsdom-global/register';

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import i18n from '~/services/i18n';

import Index from '..';

i18n.languages = ['fr'];

describe('Index', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Index i18n={i18n} t={i18n.t} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shallow render should contain Hello world', () => {
    const wrapper = shallow(<Index i18n={i18n} t={i18n.t} />);
    wrapper.contains(<div>Hello world</div>);
  });
});
