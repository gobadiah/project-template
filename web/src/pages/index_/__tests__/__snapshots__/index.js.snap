// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match snapshot 1`] = `
.emotion-3 {
  background-color: Transparent;
  border: none;
  cursor: pointer;
  outline: none;
  overflow: hidden;
  padding: 0;
  width: 220px;
  height: 100px;
}

.emotion-0 {
  color: green;
}

.emotion-8 {
  background-color: hotpink;
}

.emotion-6 {
  position: relative;
  width: 220px;
  height: 100px;
}

.emotion-1 {
  width: 100%;
}

.emotion-4 {
  visibility: hidden;
  opacity: 0;
  top: 100%;
  left: 0;
  width: 100%;
  -webkit-transform: translateY(-2em);
  -ms-transform: translateY(-2em);
  transform: translateY(-2em);
  z-index: -1;
  -webkit-transition: all 0.3s ease-in-out 0s,visibility 0s linear 0.3s,z-index 0s linear 0.01s;
  transition: all 0.3s ease-in-out 0s,visibility 0s linear 0.3s,z-index 0s linear 0.01s;
}

.emotion-7:hover .emotion-4,
.emotion-7:focus .emotion-4,
.emotion-7:focus-within .emotion-4 {
  visibility: visible;
  opacity: 1;
  z-index: 1;
  -webkit-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
  -webkit-transition-delay: 0s,0s,0.3s;
  transition-delay: 0s,0s,0.3s;
}

<div
  className="emotion-8"
>
  <div
    className="emotion-0"
  >
    Hello world
  </div>
  <div
    className="emotion-6 emotion-7"
  >
    <img
      alt="fr"
      className="emotion-1"
      src="/static/build/329f5c89d4f13938971b9b0d8d35f296.png"
    />
    <div
      className="emotion-4 emotion-5"
    >
      <button
        className="emotion-3"
        onClick={[Function]}
      >
        <img
          alt="en"
          className="emotion-1"
          src="/static/build/1b46775dab5a7245b5969a5aa84e0dff.png"
        />
      </button>
    </div>
  </div>
  <div>
    This is a pure component
  </div>
  <a
    href="/some-page"
    onClick={[Function]}
  >
    Some page
  </a>
  <a
    href="/generate-error"
    onClick={[Function]}
  >
    Generate error
  </a>
  <a
    href="/signin"
    onClick={[Function]}
  >
    Signin
  </a>
  <a
    href="/register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;
